src/check_str.c

```diff
@@ -127,9 +127,8 @@ static int percent_passed(TestStats * t)
 {
     if(t->n_failed == 0 && t->n_errors == 0)
         return 100;
-    else if(t->n_checked == 0)
```

Comment:
Are these style changes or logic changes? Are these because cppcheck indicated an issue?

As least for cppcheck 1.74 no issues were detected in the src dir. The test dir did mention the following:
```
[tests/check_check_pack.c:323] -> [tests/check_check_pack.c:328]: (warning) Either the condition 'rmsg!=0' is redundant or there is possible null pointer dereference: rmsg.
```
Which seems to indicate that there is a redundant null check. The following resolves this:
```
diff --git a/tests/check_check_pack.c b/tests/check_check_pack.c
index 7da7e57..dd5d557 100644
--- a/tests/check_check_pack.c
+++ b/tests/check_check_pack.c
@@ -325,7 +325,6 @@ START_TEST(test_ppack_onlyctx)
   ck_assert_msg (rmsg->test_line == -1,
               "Result loc line should be -1 with only CTX");
 
-  if (rmsg != NULL)
     free (rmsg);
   fclose(result_file);
 }
```

Would you mind including this into your pull request, to remove all errors/warnings before enabling the check?